pytorch_lightning/trainer/trainer.py

```diff
@@ -422,6 +422,14 @@ def __init__(
         self.optimizer_connector.on_trainer_init()
 
         # init data flags
+        if prepare_data_per_node is None:
+            prepare_data_per_node = True
+        else:
+            rank_zero_warn(
+                "Setting prepare_data_per_node with the trainer flag is deprecated and will be removed in v1.7.0!"
+                "Please use the property in the DataHooks for setting prepare_data_per_node"
```

Comment:
Users shouldn't know about DataHooks, let's be more explicit and use the public API.

Reply:
Hi, @tchaton  Thanks for the suggestion! I will update the warning. Similar to what I've asked to @yifuwang, I'm wondering if we allow user to pre-define prepare_data_per_node even when LightningDataModule is not set within trainer? Because I noticed that `data_connector` initializes with None datamodule in  `on_trainer_init`.